src/services/identity/SitesCacheService.ts

```diff
@@ -14,81 +18,232 @@ import type { GitHubRepositoryData, RepositoryData } from "@root/types/repoInfo"
 // Changes are polled at fixed intervals, stored in cache, and served
 // to avoid long load time from live querying.
 
+const DEFAULT_PAGE_PARAMS = {
+  per_page: GH_MAX_REPO_COUNT,
+  sort: "full_name",
+  page: 1,
+}
+
+type GitHubResponseHeaders = HeadersDefaults & {
+  link?: string
+}
+
+type Link = {
+  relid: "first" | "last" | "prev" | "next"
+  url: string
+  pagenum: number
+}
+
+type LinkSet = {
+  first?: Link
+  last?: Link
+  prev?: Link
+  next?: Link
+}
+
+type LinkMatch = RegExpExecArray & {
+  groups: {
+    relid: "first" | "last" | "prev" | "next"
+    url: string
+  }
+}
+
+type FetchRepoPageResult = {
+  repos: RepositoryData[]
+  links?: LinkSet
+}
+
+type CacheStore = {
+  [key: string]: RepositoryData
+}
+
+function parseGitHubLinkHeader(linkheader: string) {
```

Comment:
nit: add return type

Reply:
Hmm. the code structure in the function body makes it correctly extracted implicitly, but I'll add it anyway 👍

Reply:
Added: https://github.com/isomerpages/isomercms-backend/pull/1239/commits/033783808a41601f3ecf6d438deb3aff82cb0974